src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -795,7 +803,9 @@ rpmostree_compose_builtin_tree (int             argc,
     goto out;
 
   if (json_object_has_member (treefile, "automatic_version_prefix") &&
-      !compose_strv_contains_prefix (opt_metadata_strings, "version="))
+      !compose_strv_contains_prefix (opt_metadata_strings, "version=") &&
+      /* let --add-metadata-string=version=... take precedence */
+      (next_version == NULL))
```

Comment:
If you agree, I think then it's actually better to delete the `compose_strv_contains` version.